src/celeritas/ext/GeantImporter.cc

```diff
@@ -192,6 +192,72 @@ struct MatPropGetter
     }
 };
 
+//---------------------------------------------------------------------------//
+//! Map particles defined in \c G4MaterialConstPropertyIndex .
+std::map<std::string, PDGNumber> optical_particles_map()
+{
+    return {{"PROTON", pdg::proton()},
+            {"DEUTERON", pdg::deuteron()},
+            {"TRITON", pdg::triton()},
+            {"ALPHA", pdg::alpha()},
+            {"ION", pdg::ion()},
+            {"ELECTRON", pdg::electron()}};
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Populate an \c ImportScintComponent .
+ * To retrieve a material-only component simply do not use particle name.
+ */
+std::vector<ImportScintComponent>
+fill_vec_import_scint_comp(MatPropGetter& get_property,
+                           std::string particle_name = "")
+{
+    if (!particle_name.empty())
+    {
+        auto const map = optical_particles_map();
+        CELER_EXPECT(map.find(particle_name) != map.end());
+    }
```

Comment:
You're making a copy of the map here.
```suggestion
    CELER_EXPECT(particle_name.empty() || optical_particles_map().count(particle_name));
```